src/chromo/constants.py

```diff
@@ -107,3 +78,42 @@
     1000080160: 0.20946,  # oxygen
     1000180400: 0.00934,  # argon
 }
+
+
+def get_all_decaying_ctau():
+    all_decaying_ctau = {}
+    for p in particle.Particle.findall():
+        if (
+            (p.ctau is not None)
+            and (not math.isinf(p.ctau))
+            and (not math.isnan(p.ctau))
+            and (abs(int(p.pdgid)) < 1000000000)
+        ):
+            all_decaying_ctau[int(p.pdgid)] = p.ctau
+
+    return all_decaying_ctau
+
+
+all_decaying_ctau = get_all_decaying_ctau()
+all_decaying_pids = [pid for pid in all_decaying_ctau]
+
+
+def long_lived_for(tau0, mm=False):
+    stable = []
+    if mm:
+        ctau0 = tau0
+    else:
+        ctau0 = tau0 * c * 1e1  # in mm
+
+    for pid, ctau in all_decaying_ctau.items():
+        if ctau > ctau0:
+            stable.append(pid)
+    return stable
+
+
+# Default definition of final state particle
+# All particles with proper lifetime shorter than this
+# will decay
+tau_stable = 30e-12  # 30 ps, typical value at LHC
+# standard long-lived particles with life-time > 30 ps
+long_lived = tuple(long_lived_for(tau_stable))
```

Comment:
Remove all this. There is only a countable set of long-lived SM particles, which are the ones in the tuple `long_lived`. If you don't trust that constant, you can add a unit test which compares our list with one computed from `particle`.

Reply:
I see that you want to enforce that the sequence `long_lived` corresponds to the `tau_stable` value. But we don't actually ever want to change this, so there is no need to make this dynamic.

I don't think that `get_all_decaying_ctau`, `all_decaying_ctau`, and `all_decaying_pids` should be part of the public API. These look like implementation details to me that you need to implement the Pythia8Decay handler. Therefore, you should put these into `decay_handler.py`

Reply:
If you want to test against `all_decaying_pids`, you probably want this to be a numpy array or a set, depending on the use case, not a list. A search in a list costs linear time, but only logarithmic time in a set.

Reply:
I moved `long_lived_for` to util. It is still required to get all decaying particles. All generators has their own default settings. To have full control on decaying/stable particles, all known decaying particles should be set unstable, before setting some subset as stable. 

`long_lived_for` can accept `tau` to return only subset of decaying particles that are stable for `tau` sec/mm.

`_decaying_pids` changed to sets for searching of pids in set_stable.

Reply:
One can set new tau condition like this:

```python
generator.final_state_particles = long_lived_for(tau)
```

All unstable particles will be set as stable:

```python
generator.final_state_particles = long_lived_for()
```

All unstable particles will decay if:

```python
generator.final_state_particles = []
```

Reply:
Ok, if you want to have a function that computes the number of long-lived particles depending on tau, then such a function should live in a module called "constants". It can be a in `util.py`. The list in `constants.py` should refer to the LPCC MB+UE working group's definition of primary particles, which uses tau > 30 ps. ALICE wrote a nice document about this https://inspirehep.net/literature/1945324. It is true that ATLAS considers only particles with tau > 300 ps as primary, but the ATLAS definition makes no sense, since it considers prompt hyperons as non-primary.

I want us to use the LPCC MB+UE working group's definition of primary particles by default. If we then provide a function that computes the list of long-lived particles for a different tau in `util.py` everyone is covered, no?

Reply:
So `long_lived_for` is in `util.py`?

Reply:
Agree with 30ps as standard!

A function and some readable docstring would be good. And, there should be a TODO in the source code reminding us to write about how to set stable particles in a proper documentation. Since we don't have any config file that can be browsed by a curious used, so we need to rely more and more on docs due to complexity.